### PR TITLE
feat(v2): JAX-native MBN2014 activation parameterization (issue #56)

### LINKED ISSUE #56: feat(v2): JAX-native MBN2014 activation parameterization
## Context

PR #55 (closes #36) shipped the JAX-native ARG2000 activation scheme.
MBN2014 was explicitly listed in #36 but was deferred because it requires
non-trivial JAX infrastructure that ARG2000 does not:

- The bisection loop over smax must be replaced by `jax.lax.while_loop`
  so the computation is JAX-traceable.
- Differentiability through the root-finding requires `jax.lax.custom_vjp`
  implementing the implicit function theorem:
  `dsmax/dθ = -(∂F/∂θ) / (∂F/∂smax)` where `F(smax, θ) = 0`.

The legacy NumPy implementation lives at `pyrcel.legacy.activation.mbn2014`
and is still accessible via `pyrcel.activation.mbn2014` for non-differentiable use.

## Tasks

- [ ] Implement `jax.lax.while_loop` bisection in `pyrcel/activation/_mbn2014.py`
- [ ] Add `jax.lax.custom_vjp` for the implicit-function-theorem gradient
- [ ] `MBN2014` class satisfying `ActivationScheme` interface
- [ ] Export from `pyrcel/activation/__init__.py`
- [ ] Equivalence tests vs `pyrcel.legacy.activation.mbn2014`
- [ ] Adjoint/FD gradient cross-check (at least for `dsmax/dV`, `dsmax/dNs`)

## Notes

- The `_sintegral` inner function in the legacy code contains Python-level
  branches (`if critical`) that depend on the value of `smax`; these must
  be converted to `jax.lax.cond` calls.
- The `_erfp` polynomial approximation can be replaced by `jax.scipy.special.erfc`.
- An intermediate milestone (acceptable if full differentiation is too costly):
  a `jax.pure_callback` shim around the NumPy implementation, flagged
  non-differentiable in the docstring.

### PR DESCRIPTION
**Pull request by darothen**
## Summary

- `pyrcel/activation/_mbn2014.py` — full JAX re-implementation of the Morales Betancourt & Nenes (2014) scheme
- `jax.lax.while_loop` bisection finds smax (JIT-compilable, O(log(1/tol)) iterations)
- Differentiability via **implicit function theorem** (`jax.custom_vjp`): ∂smax/∂θ = −(∂F/∂θ)/(∂F/∂smax) evaluated by a single `jax.grad` call on the algebraic residual at the converged root — no need to differentiate through the bisection loop
- `jnp.where` for the critical/non-critical population-splitting branch (scalar flag, vectorised over modes)
- Polynomial helpers (`_vpres_jax`, `_erfp_jax`) reproduce the legacy Nenes-group constants exactly for numerical agreement
- `MBN2014` class satisfies the `ActivationScheme` interface
- The legacy `mbn2014` re-export from `pyrcel.activation` is replaced by the JAX implementation; the NumPy original remains accessible at `pyrcel.legacy.activation.mbn2014`

## Test plan

- [x] 12 cross-check tests (3 aerosol scenarios × 4 updraft conditions): smax within 1e-5 relative of legacy
- [x] Activated fractions within 1e-4 relative of legacy
- [x] Non-unity accom correction vs legacy
- [x] 5 gradient tests (V, T, P, Ns, kappas): IFT grad vs central FD within 1e-2 relative
- [x] Physical invariants: smax monotone in V, fracs ∈ [0,1], smax > 0
- [x] `MBN2014` class interface and public import

Closes #56

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Swapping the default `mbn2014` export changes behavior for callers that expected NumPy/SciPy; JAX bisection and IFT gradients add numerical edge cases, though tests heavily guard legacy parity.
> 
> **Overview**
> Adds a **JAX-native MBN2014** droplet activation scheme alongside ARG2000, exposed as `mbn2014` / `MBN2014` from `pyrcel.activation` and lazy `pyrcel.MBN2014`. Imports of `mbn2014` from `pyrcel.activation` now resolve to the JAX implementation; the NumPy reference stays at `pyrcel.legacy.activation.mbn2014`.
> 
> **smax** is found by JIT-friendly `jax.lax.while_loop` bisection on an implicit residual; backward mode uses **`jax.custom_vjp`** and the implicit function theorem so `jax.grad` works w.r.t. V, T, P, and aerosol inputs without differentiating through the loop. Legacy-matched polynomial helpers and vectorised population-splitting integrals keep numerical agreement with the legacy oracle.
> 
> New **`tests/test_mbn2014.py`** cross-checks smax and activated fractions vs legacy, validates IFT gradients against finite differences, checks physical monotonicity/bounds, and covers the `MBN2014` class and public imports.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ecfd326fc482fe39c171c9e289ff786ae12dc1e3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by darothen** [comment]
@cursor review once

File: pyrcel/activation/_mbn2014.py [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+The scheme finds the maximum parcel supersaturation by solving the implicit
+equation F(smax, θ) = 0 via bisection, then computes per-mode activated
+fractions from lognormal statistics.  The bisection uses
+:func:`jax.lax.while_loop` so it is JIT-compilable but not differentiable
+through the loop itself.  Differentiability w.r.t. the physical inputs is
+recovered by the **implicit function theorem** (IFT), implemented via
+:func:`jax.custom_vjp`:
+
+    ∂smax/∂θ  =  −(∂F/∂θ) / (∂F/∂smax)   at  F(smax, θ) = 0
+
```
Comment (darothen) [inline pyrcel/activation/_mbn2014.py:16]: Re-write using latex or mathjax

File: pyrcel/activation/_mbn2014.py [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+jax.config.update("jax_enable_x64", True)
+
+import jax.numpy as jnp  # noqa: E402
+from jax import Array  # noqa: E402
+from jax.typing import ArrayLike  # noqa: E402
+
+from .. import constants as c  # noqa: E402
+from ..thermo import ka_cont  # noqa: E402
+from ._arg2000 import _lognormal_act  # noqa: E402
```
Comment (darothen) [inline pyrcel/activation/_mbn2014.py]: Consider re-factoring this (and related utility functions) to a common activation module.

File: pyrcel/activation/_mbn2014.py [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+    P: Array,
+    mus_m: Array,
+    sigmas: Array,
+    Ns_m: Array,
+    kappas: Array,
+    accom: Array,
+) -> Array:
+    """Implicit equation F(smax, θ) = 0 whose root defines smax.
+
+    All operations are standard JAX, making this fully differentiable w.r.t.
```
Comment (darothen) [inline pyrcel/activation/_mbn2014.py]: I don't use the construction "w.r.t", use the full phrase "with respect to" anywhere this is employed.

**Review by darothen — COMMENTED** [review]
Some changes

File: pyrcel/activation/_mbn2014.py (hunk 1 above)
Comment (darothen) [reply pyrcel/activation/_mbn2014.py:16]: Done — rewrote the IFT equation in RST math block using LaTeX notation.

File: pyrcel/activation/_mbn2014.py (hunk 2 above)
Comment (darothen) [reply pyrcel/activation/_mbn2014.py]: Done — extracted `_lognormal_act` into `pyrcel/activation/_common.py`. Both `_arg2000.py` and `_mbn2014.py` now import from the shared module. `_kohler_crit_approx` stays in `_arg2000.py` as it is not shared.

File: pyrcel/activation/_mbn2014.py (hunk 3 above)
Comment (darothen) [reply pyrcel/activation/_mbn2014.py]: Done — replaced all three occurrences of "w.r.t." with "with respect to" throughout the file.
